src/currentprocess.rs

```diff
@@ -279,55 +252,76 @@ impl Default for OSProcess {
 }
 
 // ------------ test process ----------------
+
 #[cfg(feature = "test")]
-#[derive(Clone, Debug, Default)]
 pub struct TestProcess {
-    pub cwd: PathBuf,
-    args: Vec<String>,
-    vars: HashMap<String, String>,
-    stdin: filesource::TestStdinInner,
-    stdout: filesource::TestWriterInner,
-    stderr: filesource::TestWriterInner,
+    pub process: Process,
+    #[allow(dead_code)] // guard is dropped at the end of the test
+    guard: DefaultGuard,
```

Comment:
I guess the linter actually wants you to write `_guard`, see https://github.com/search?q=org%3Arust-lang%20_guard%3A&type=code for more examples across the org.